### PR TITLE
Feature/MainPage

### PR DESCRIPTION
**Pull request by slowdata**
Main Page basic build.

It has a Header and list of Items to display.
(Items will be fetch with Axios from the API if DB has Data)

### CODE REVIEW

File: src/Components/App.js [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+  constructor(props) {
+    super(props);
+    this.state = {
+      items: []
+    };
+  }
+
+  async componentDidMount() {
+    try {
+      const items = (await Axios.get(`${URI}items`)).data;
```
Comment (uvluni) [inline src/Components/App.js:20]: Nice!

File: src/Components/Layout/Header.js [hunk 2]
```diff
@@ -0,0 +1,48 @@
+import React from "react";
+import PropTypes from "prop-types";
+import { withStyles } from "material-ui/styles";
+import AppBar from "material-ui/AppBar";
+import Toolbar from "material-ui/Toolbar";
+import Typography from "material-ui/Typography";
+import Button from "material-ui/Button";
+
+const styles = {
```
Comment (uvluni) [inline src/Components/Layout/Header.js]: Const styles should be imported from the component folder, from a file named styles.css.

File: src/Components/ListItems/Item.js [hunk 3]
```diff
@@ -0,0 +1,46 @@
+import React from "react";
+import PropTypes from "prop-types";
+import { Card, Typography, Button } from "material-ui";
+import { CardMedia, CardContent, CardActions } from "material-ui/Card";
+import { withStyles } from "material-ui/styles";
+
+const styles = {
```
Comment (uvluni) [inline src/Components/ListItems/Item.js]: Const styles should be imported from the component folder, from a file named styles.css.

File: src/Components/ListItems/Item.js [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+  media: {
+    height: 200
+  }
+};
+
+const Item = ({ classes, item }) => (
+  <Card className={classes.card} raised={true}>
+    <CardMedia
+      className={classes.media}
+      //image="https://images.pexels.com/photos/574069/pexels-photo-574069.jpeg?w=940&h=650&auto=compress&cs=tinysrgb"
```
Comment (uvluni) [inline src/Components/ListItems/Item.js]: When done, delete comments and console.log

File: src/Components/SearchBar/index.js [hunk 5]
```diff
@@ -0,0 +1,43 @@
+import React from "react";
+import { Grid, Paper, TextField, Button } from "material-ui";
+
+const styles = {
```
Comment (uvluni) [inline src/Components/SearchBar/index.js]: Const styles should be imported from the component folder, from a file named styles.css.

**Review by uvluni — APPROVED** [review]
I'm not requesting changes so we will be able to merge and continue working, but these are my pointers:
1. Const styles should be imported from the component folder, from a file named styles.css.
2. List and Item components should be in separated folders (maybe nested inside one folder, it's up to you).
I see you export the index file to import and export ListItems and for me, it feels complicated to read.
I don't think we need index for each component...
3. I liked your convention of naming the component file like the component name (Item.js, List.js, etc...).
So Search bar should be SearchBar.js
4. When done, delete comments and console.log

Love what you did, it looks awesome.
LMK if you disagree with any of my points so we will be able to set the conventions for the rest of the project

File: src/Components/App.js (hunk 1 above)
Comment (slowdata) [reply src/Components/App.js:20]: Thkx. I really appreciate your opinion. :)

File: src/Components/ListItems/Item.js (hunk 3 above)
Comment (slowdata) [reply src/Components/ListItems/Item.js]: I agree with you. Just so you know what I was thinking is this:
-When using only a simple style change I would set it Inline.
-If it was more than one type or even more than one style applied to one component I would set a const object.
-If it needed more I would go to a separated file.

The main reason was to better reading of the code and better understanding the code if it where in the same file -- Like in a vue file;

File: src/Components/ListItems/Item.js (hunk 4 above)
Comment (slowdata) [reply src/Components/ListItems/Item.js]: I think I didn't check my code. It worked so it when't strait to github (sorry)

**Comment by slowdata** [comment]
Completely agree with you!

- Also to explain your **2nd point** - My idea what if I needed to import more than one component of the same "family" I would only need to import the folder because by default it would go to **index.js**.
As you can see index only as one component so it doesn't make much sense to do that. :)

Tell me what you think I should do. Make the changes and make a new PR? Don't know I we should proceed with that.

**Comment by slowdata** [comment]
I made the changes as agreed. Will close this PR and make a new one

**Pull request closed without merging**